seqr/views/utils/dataset_utils.py

```diff
@@ -123,16 +124,21 @@ def _update_samples_for_dataset(
         ))
 
     not_loaded_samples = []
+    update_json = {}
+    if dataset_name:
+        update_json['dataset_name'] = dataset_name
+    if elasticsearch_index:
+        update_json['elasticsearch_index'] = elasticsearch_index
     for sample_id, sample in matched_sample_id_to_sample_record.items():
+        sample_update_json = {}
         if sample_dataset_path_mapping and sample_dataset_path_mapping.get(sample_id):
-            sample.dataset_file_path = sample_dataset_path_mapping[sample_id]
-        if dataset_name:
-            sample.dataset_name = dataset_name
+            sample_update_json['dataset_file_path'] = sample_dataset_path_mapping[sample_id]
         if sample.sample_status != Sample.SAMPLE_STATUS_LOADED:
             not_loaded_samples.append(sample_id)
-            sample.sample_status = Sample.SAMPLE_STATUS_LOADED
-            sample.loaded_date = timezone.now()
-        sample.save()
+            sample_update_json['sample_status'] = Sample.SAMPLE_STATUS_LOADED
+            sample_update_json['loaded_date'] = timezone.now()
```

Comment:
would run timezone.now() once before the loop, so that all the samples get the same timestamp